### ISSUE TITLE
Formulas containing $ get half-displayed as text

### ISSUE THREAD

**Issue by SanJacobs** [tgrosinger/table-editor-obsidian#224]
**Describe the bug**
Parts of formulas past any $-signs get evaluated by Obsidian as if it was normal text, rendering markdown formatting characters as text styling.

This formula:

    

Displays like this, when not in source mode:
![image](https://user-images.githubusercontent.com/21090839/204283791-beb438af-4655-46a4-9ece-125fe49e4eaf.png)

And like this, with ligatures enabled on mobile:
![image](https://user-images.githubusercontent.com/21090839/204285056-1589c69a-00c3-46e2-be34-2241b6fd932c.png)

This is of course expected behavior, given that this is just text in a markdown document.

However, a really easy fix would be if the formulas could be written with the ending arrow containing an exclamation point (`--!>`), then they seem to display correctly/less-confusingly.

![image](https://user-images.githubusercontent.com/21090839/204285980-9119c9ed-8299-4761-94e0-0bc362da025b.png)
<sup>(I also happen to find this more visually appealing, because it is symmetrical)</sup>

When doing that, though, they aren't recognized by the plugin as a formula.

**To Reproduce**
1. Write formula containing $ and markdown formatting characters

**Expected behavior**
I expect to see every symbol I have written, and for them to be styled the same, because in this context they are not formatting characters, they are functions.

**Desktop:**
 - OS: Windows 10
 - Obsidian Version: 1.0.3
 - Advanced Tables Version: 0.18.0
 
**Mobile:**
 - OS: Android
 - Obsidian Version: 1.4.1
 - Advanced Tables Version: 0.18.0

**Comment by tgrosinger** [comment]
That is an interesting idea, however it would then no longer be a valid HTML comment, so the formula might be displayed in some rendering engines.

https://developer.mozilla.org/en-US/docs/Web/API/Comment

I'll have to think about this some more.

**Comment by SanJacobs** [comment]
True. Of course, this probably shouldn't be a forced transition from `-->` to `--!>`. However, adding it as a recognized alternative for us users who don't care about the rendering outside of obsidian would be great.

**Comment by AlexLJordan** [comment]
Hi just wanted to chip in: I'm having the following issue

This line is provoking an error message:
`<!-- TBLFM: @I$>..@>$>=sum($2..$7) -->`

It get's highlighted like this:

![Pasted image 20231118010403](https://github.com/tgrosinger/advanced-tables-obsidian/assets/10133257/65504f9a-8aea-426a-9fa4-6c05cb108045)

When I hover over it, the tooltip says "Misplaced @".

I tried CSS to stop rendering comments in live-preview, but that just removes the part until the first $ symbol:

![Pasted image 20231118010829](https://github.com/tgrosinger/advanced-tables-obsidian/assets/10133257/d8b81c4b-16c5-483c-af44-6d34bd3f01e9)

I'd really appreciate some help here, this bright yellow highlighting is super annoying.
